server/src-lib/Hasura/RQL/DDL/Schema/Cache.hs

```diff
@@ -543,7 +543,7 @@ withMetadataCheck cascade action = do
   where
     reportFuncs = T.intercalate ", " . map dquoteTxt
 
-    processSchemaChanges :: (MonadTx m, CacheRM m) => SchemaDiff -> m ()
+    processSchemaChanges :: (MonadTx m, CacheRWM m) => SchemaDiff -> m ()
```

Comment:
Don't change this to `CacheRWM`! Preserve `CacheRM`. The `(MonadTx m, CacheRM m)` implies the function is extracting metadata cache from the state and modifying the necessary metadata in the database. At the end, we'd call `buildSchemaCache` to update the metadata cache anyway and check for any new inconsistency (See `withMetadataCheck` function).

Reply:
I wanted to use the `runUpdateRemoteRelationship` function for the validation it does, but after your comment I realized that, that check will be done in `withMetadataCheck` again. 
I have modified the code to use `CacheRM`.